### PR TITLE
DUPLO-8848 - using new v3 rds api

### PR DESCRIPTION
**Pull request by pravin-duplocloud**
DUPLO-8848 - update rds v3 api. refactoring and cleanup be. use new v3 api  in ui/tf

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-8848 update TF to use newer RDS routes](https://app.clickup.com/t/8655600/DUPLO-8848)

File: docs/resources/rds_instance.md [hunk 1]
```diff
@@ -111,5 +111,5 @@ Import is supported using the following syntax:
 #  - *TENANT_ID* is the tenant GUID
 #  - *SHORTNAME* is the short name of the database (without the duplo prefix)
 #
-terraform import duplocloud_rds_instance.mydb v2/subscriptions/*TENANT_ID*/RDSDBInstance/*SHORTNAME*
+terraform import duplocloud_rds_instance.mydb v3/subscriptions/*TENANT_ID*/RDSDBInstance/*SHORTNAME*
```
Comment (joek-duplo) [inline docs/resources/rds_instance.md:114]: revert this change

File: docs/resources/rds_read_replica.md [hunk 2]
```diff
@@ -88,5 +88,5 @@ Import is supported using the following syntax:
 #  - *TENANT_ID* is the tenant GUID
 #  - *SHORTNAME* is the short name of the database read replica (without the duplo prefix)
 #
-terraform import duplocloud_rds_read_replica.read_replica v2/subscriptions/*TENANT_ID*/RDSDBInstance/*SHORTNAME*
+terraform import duplocloud_rds_read_replica.read_replica v3/subscriptions/*TENANT_ID*/RDSDBInstance/*SHORTNAME*
```
Comment (joek-duplo) [inline docs/resources/rds_read_replica.md:91]: revert this change

File: duplocloud/resource_duplo_rds_instance.go [hunk 3]
```diff
@@ -249,7 +249,7 @@ func resourceDuploRdsInstanceRead(ctx context.Context, d *schema.ResourceData, m
 	for key := range jo {
 		d.Set(key, jo[key])
 	}
-	d.SetId(fmt.Sprintf("v2/subscriptions/%s/RDSDBInstance/%s", duplo.TenantID, duplo.Name))
+	d.SetId(fmt.Sprintf("v3/subscriptions/%s/RDSDBInstance/%s", duplo.TenantID, duplo.Name))
```
Comment (joek-duplo) [inline duplocloud/resource_duplo_rds_instance.go:252]: revert this change

File: duplocloud/resource_duplo_rds_instance.go [hunk 4]
```diff
@@ -268,7 +268,7 @@ func resourceDuploRdsInstanceCreate(ctx context.Context, d *schema.ResourceData,
 	// Populate the identifier field, and determine some other fields
 	duplo.Identifier = duplo.Name
 	tenantID := d.Get("tenant_id").(string)
-	id := fmt.Sprintf("v2/subscriptions/%s/RDSDBInstance/%s", tenantID, duplo.Name)
+	id := fmt.Sprintf("v3/subscriptions/%s/RDSDBInstance/%s", tenantID, duplo.Name)
```
Comment (joek-duplo) [inline duplocloud/resource_duplo_rds_instance.go:271]: revert this change

File: duplocloud/resource_duplo_rds_read_replica.go [hunk 5]
```diff
@@ -161,7 +161,7 @@ func resourceDuploRdsReadReplicaRead(ctx context.Context, d *schema.ResourceData
 	for key := range jo {
 		d.Set(key, jo[key])
 	}
-	d.SetId(fmt.Sprintf("v2/subscriptions/%s/RDSDBInstance/%s", duplo.TenantID, duplo.Name))
+	d.SetId(fmt.Sprintf("v3/subscriptions/%s/RDSDBInstance/%s", duplo.TenantID, duplo.Name))
```
Comment (joek-duplo) [inline duplocloud/resource_duplo_rds_read_replica.go:164]: revert this change

File: duplocloud/resource_duplo_rds_read_replica.go [hunk 6]
```diff
@@ -190,7 +190,7 @@ func resourceDuploRdsReadReplicaCreate(ctx context.Context, d *schema.ResourceDa
 	duplo.Identifier = duplo.Name
 	duplo.Engine = duploWriterInstance.Engine
 	duplo.Cloud = duploWriterInstance.Cloud
-	id := fmt.Sprintf("v2/subscriptions/%s/RDSDBInstance/%s", tenantID, duplo.Name)
+	id := fmt.Sprintf("v3/subscriptions/%s/RDSDBInstance/%s", tenantID, duplo.Name)
```
Comment (joek-duplo) [inline duplocloud/resource_duplo_rds_read_replica.go:193]: revert this change

File: examples/resources/duplocloud_rds_instance/import.sh [hunk 7]
```diff
@@ -2,4 +2,4 @@
 #  - *TENANT_ID* is the tenant GUID
 #  - *SHORTNAME* is the short name of the database (without the duplo prefix)
 #
-terraform import duplocloud_rds_instance.mydb v2/subscriptions/*TENANT_ID*/RDSDBInstance/*SHORTNAME*
```
Comment (joek-duplo) [inline examples/resources/duplocloud_rds_instance/import.sh:5]: revert this change

**Review by joek-duplo — CHANGES_REQUESTED** [review]

File: duplocloud/resource_duplo_rds_read_replica.go (hunk 6 above)
Comment (pravin-duplocloud) [reply duplocloud/resource_duplo_rds_read_replica.go]: done

File: duplocloud/resource_duplo_rds_read_replica.go (hunk 5 above)
Comment (pravin-duplocloud) [reply duplocloud/resource_duplo_rds_read_replica.go]: done

File: duplocloud/resource_duplo_rds_instance.go (hunk 4 above)
Comment (pravin-duplocloud) [reply duplocloud/resource_duplo_rds_instance.go]: done

File: duplocloud/resource_duplo_rds_instance.go (hunk 3 above)
Comment (pravin-duplocloud) [reply duplocloud/resource_duplo_rds_instance.go]: done

File: docs/resources/rds_instance.md (hunk 1 above)
Comment (pravin-duplocloud) [reply docs/resources/rds_instance.md]: done

File: docs/resources/rds_read_replica.md (hunk 2 above)
Comment (pravin-duplocloud) [reply docs/resources/rds_read_replica.md]: done

File: examples/resources/duplocloud_rds_instance/import.sh (hunk 7 above)
Comment (pravin-duplocloud) [reply examples/resources/duplocloud_rds_instance/import.sh:5]: done

File: duplocloud/resource_duplo_rds_read_replica.go [hunk 8]
```diff
@@ -161,7 +161,7 @@ func resourceDuploRdsReadReplicaRead(ctx context.Context, d *schema.ResourceData
 	for key := range jo {
 		d.Set(key, jo[key])
 	}
-	d.SetId(fmt.Sprintf("v2/subscriptions/%s/RDSDBInstance/%s", duplo.TenantID, duplo.Name))
+	d.SetId(fmt.Sprintf("v3/subscriptions/%s/aws/rds/instance/%s", duplo.TenantID, duplo.Name))
```
[the remark this answers is not in the document]
Comment (pravin-duplocloud) [reply duplocloud/resource_duplo_rds_read_replica.go]: done

**Review by joek-duplo — APPROVED** [review]

File: duplosdk/rds_instance.go [hunk 9]
```diff
@@ -84,37 +84,48 @@ func (c *Client) RdsInstanceUpdate(tenantID string, duploObject *DuploRdsInstanc
...
 		return nil, err
 	}
 	return &rp, err
+
 }
 
 // RdsInstanceDelete deletes an RDS instance via the Duplo API.
 func (c *Client) RdsInstanceDelete(id string) (*DuploRdsInstance, ClientError) {
-	idParts := strings.SplitN(id, "/", 5)
+	idParts := strings.Split(id, "/")
```
Comment (joek-duplo) [inline duplosdk/rds_instance.go:121]: revert this, the ID is not changing

File: duplosdk/rds_instance.go [hunk 10]
```diff
@@ -84,37 +84,48 @@ func (c *Client) RdsInstanceUpdate(tenantID string, duploObject *DuploRdsInstanc
...
 	tenantID := idParts[2]
-	name := idParts[4]
+	name := idParts[len(idParts)-1]
```
Comment (joek-duplo) [inline duplosdk/rds_instance.go:123]: revert this the ID is not changing

File: duplosdk/rds_instance.go [hunk 11]
```diff
@@ -126,20 +137,19 @@ func (c *Client) RdsInstanceDelete(id string) (*DuploRdsInstance, ClientError) {
 
 // RdsInstanceGet retrieves an RDS instance via the Duplo API.
 func (c *Client) RdsInstanceGet(id string) (*DuploRdsInstance, ClientError) {
-	idParts := strings.SplitN(id, "/", 5)
+	idParts := strings.Split(id, "/")
```
Comment (joek-duplo) [inline duplosdk/rds_instance.go:140]: revert this, the ID is not changing

File: duplosdk/rds_instance.go [hunk 12]
```diff
@@ -126,20 +137,19 @@ func (c *Client) RdsInstanceDelete(id string) (*DuploRdsInstance, ClientError) {
...
 	tenantID := idParts[2]
-	name := idParts[4]
+	name := idParts[len(idParts)-1] // use new v3 api ..take last one
```
Comment (joek-duplo) [inline duplosdk/rds_instance.go:142]: revert this, the ID is not changing

**Review by joek-duplo — CHANGES_REQUESTED** [review]
still some bugs left

File: duplosdk/rds_instance.go (hunk 9 above)
Comment (pravin-duplocloud) [reply duplosdk/rds_instance.go:121]: done

File: duplosdk/rds_instance.go (hunk 10 above)
Comment (pravin-duplocloud) [reply duplosdk/rds_instance.go:123]: done

File: duplosdk/rds_instance.go (hunk 11 above)
Comment (pravin-duplocloud) [reply duplosdk/rds_instance.go:140]: done

File: duplosdk/rds_instance.go (hunk 12 above)
Comment (pravin-duplocloud) [reply duplosdk/rds_instance.go:142]: done

**Review by joek-duplo — APPROVED** [review]
please have QA test before merging

**Pull request merged**